Server/src/main/java/org/openas2/util/AS2Util.java

```diff
@@ -147,6 +134,16 @@ public static void parseMDN(Message msg, X509Certificate receiver) throws OpenAS
         }
     }
 
+    private static String getMimeBodyPartText(MimeBodyPart part) throws MessagingException, IOException {
+        Object content = part.getContent();
+
+        if (InputStream.class.isAssignableFrom(content.getClass())) {
+            return IOUtils.toString((InputStream) content, StandardCharsets.UTF_8);
```

Comment:
Not sure how to get the actual charset ...

Reply:
After line 119 of parseMDN method add this:
```
                Charset charset = StandardCharsets.UTF_8;
                String charsetFromContentType = reportType.getParameter("charset");
                if (charsetFromContentType != null) {
                	try {
						charset = Charset.forName(charsetFromContentType);
					} catch (Exception e) {
						// Just warn and allow the default to go thru
						logger.warn("Received charset string that cannot be parsed: " + charsetFromContentType + msg.getLogMsgID());
					}
                }
```
Then pass `charset` into the getMimeBodyPartText method.